Source/JavaScriptCore/runtime/JSObject.h

```diff
@@ -959,7 +961,7 @@ class JSObject : public JSCell {
     // indexing should be sparse or because we're having a bad time.
     ContiguousJSValues tryMakeWritableContiguous(VM& vm)
     {
-        if (LIKELY(hasContiguous(indexingType()) && !isCopyOnWrite(indexingMode())))
+        if (LIKELY(hasAnyContiguous(indexingType()) && !isCopyOnWrite(indexingMode())))
             return m_butterfly->contiguous();
             
         return tryMakeWritableContiguousSlow(vm);
```

Comment:
Is this doing what we expect in arrayify?

Reply:
That is a really nice catch, thank you. Fixed `tryMakeWritableContiguous` to return an empty values for new indexing type.